File: day1.py
def reader(folderName) :
    with open(folderName,encoding="utf-8") as f: 
        data = f.read()
    return data 

input_day_one = reader("day1.txt")

print(type(input_day_one))

print(input_day_one)


def day_one(input) :
    counter = 0
    for ch in input:
        if(ch == "("): 
            counter+=1 
        elif (ch == ")"): 
            counter-=1 
        else: 
            continue 
    return counter 


resultDayOne = day_one(input_day_one)

print(type(resultDayOne))
print(resultDayOne)

File: js/day1.js
import fs from "fs";

const reader = (input) => {
  const esraAndMert = fs.readFileSync(input, "utf-8");
  return esraAndMert;
};

const day1 = (text) => {
  let counter = 0;
  for (let i = 0; i < text.length; i++) {
    if (text[i] == "(") counter++;
    else counter--;
  }
  return counter;
};

console.log(day1(reader("../day1.txt")));
